### PR TITLE
implementation of running a lambda in ecs

### PR DESCRIPTION
**Pull request by sethvincent**
This is based in part on https://git.earthdata.nasa.gov/projects/CUMULUS/repos/ecs-lambda-runner/browse

Updated to work with step function tasks using `sf.getActivityTask` and related step functions API methods.

### CODE REVIEW

**Comment by sethvincent** [comment]
When given enough time, ecs service creation will fail with:

```
Service arn:aws:ecs:us-east-1:158242077916:service/sdv-say-hello-cumulus-SayHelloNodeEcsECSService-PQZHC6TJ46UO did not stabilize.
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    docker:
+      - image: docker:17.12.1-ce
+    working_directory: ~/ecs-lambda-runner
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: |
+          docker build -t developmentseed/cumulus:lambda-runner .
+          docker login -u scisco -p $DOCKER_PASSWORD
+          docker push developmentseed/cumulus:lambda-runner
```
Comment (scisco) [inline .circleci/config.yml]: let's push this to the `cumuluss` org on docker hub?

Comment (scisco) [reply .circleci/config.yml]: to be exact, push it to: `cumuluss/cumulus-ecs-task`

Comment (scisco) [reply .circleci/config.yml]: Also, what is our plan for versioning here? I think we should not release with `latest` tag and start versioning this image.

**Comment by scisco** [comment]
@sethvincent the service ARN is probably not stabilized because it lacks the correct permissions. You have to make sure that your ec2 instance has the right to download the zip file of a given lambda.

Also you have to make sure the base image has the latest version of python 2.7

**Comment by sethvincent** [comment]
This branch can now work with a couple of outside changes:

- changing cumulus-message-adapter-js to use an env var to specify the cumulus-message-adapter location
- changing iams with `- states:SendTaskHeartbeat` in the EcsRole allow actions

**Comment by sethvincent** [comment]
I've updated this with:
- circleci improvements that publish to npm and docker hub when a PR gets merged that bumps the version in package.json
- revises the dockerfile
- adds .eslintrc.json from cumulus-nasa/cumuluse
- README.md updates with instructions that will help with reviewing and testing this PR
- a bunch of lint and bug fixes and a slight refactor from when I first opened this PR

**Pull request merged**